gcc/rust/util/rust-poly.h

```diff
@@ -0,0 +1,111 @@
+/* An implementation of polymophic_value for C++11.
+   https://github.com/jbcoe/polymorphic_value */
```

Comment:
is your implementation sufficiently different from the reference that it won't cause copyright assignment issues?

Reply:
Currently it looks quite similar, but as discussed earlier this implementation has a runtime overhead. I might end up rewriting it again to solve the performance problem(which should make it distinct enough).